web-common/src/features/charts/render/vega-tooltip.ts

```diff
@@ -0,0 +1,54 @@
+import {
+  mouseLocationToBoundingRect,
+  placeElement,
+} from "@rilldata/web-common/lib/place-element";
+import { VLTooltipFormatter } from "../types";
+
+const TOOLTIP_ID = "rill-vg-tooltip";
+
+export class VegaLiteTooltipHandler {
+  location = "left";
+  alignment = "middle";
+  distance = 0;
+  pad = 8;
+  public valueFormatter: VLTooltipFormatter;
+
+  constructor(valueFormatter) {
+    this.valueFormatter = valueFormatter;
+    this.handleTooltip = this.handleTooltip.bind(this);
```

Comment:
You can skip this and just make handleTooltip an arrow function, which will also fix the linting errors when this method is used.